coordinator/cluster/cluster.go

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package cluster
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-load-test-ng/coordinator/agent"
+)
+
+// LoadAgentCluster is the object holding information about all the load-test
+// agents available in the cluster.
+type LoadAgentCluster struct {
+	config LoadAgentClusterConfig
+	agents []*agent.LoadAgent
+}
+
+// New creates and initializes a new LoadAgentCluster for the given config.
+// An error is returned if the initialization fails.
+func New(config LoadAgentClusterConfig) (*LoadAgentCluster, error) {
+	if ok, err := config.IsValid(); !ok {
+		return nil, err
+	}
+	agents := make([]*agent.LoadAgent, len(config.Agents))
+	for i := 0; i < len(agents); i++ {
+		agent, err := agent.New(config.Agents[i])
+		if err != nil {
+			return nil, fmt.Errorf("cluster: failed to create agent \n%w", err)
+		}
+		agents[i] = agent
+	}
+	return &LoadAgentCluster{
+		agents: agents,
+		config: config,
+	}, nil
+}
+
+// Run starts all the load-test agents available in the cluster.
+func (c *LoadAgentCluster) Run() error {
+	for _, agent := range c.agents {
+		err := agent.Start()
+		if err != nil {
```

Comment:
if we abort on first error, probably should shutdown the already started agents

Reply:
Yes, I like the idea of leaving the state as we found it so it makes sense to shutdown if an agent fails to start especially since we currently exit the program when that happens. That said, I'd prefer to leave this decision to the `coordinator` package.